# HG changeset patch
# User Luca Greco <lgreco@mozilla.com>
# Date 1458675228 -3600
#      Tue Mar 22 20:33:48 2016 +0100
# Node ID 43db29b0a05f3562829cc958631593b32ad6c914
# Parent  af62f63415ba630411ef8038e3304b00ef1be8ff
Bug 1211665 - [webext] Mark content scripts sandboxes with the extension id. r?kmag

MozReview-Commit-ID: IT0M4wzfpAQ

diff --git a/toolkit/components/extensions/ExtensionContent.jsm b/toolkit/components/extensions/ExtensionContent.jsm
--- a/toolkit/components/extensions/ExtensionContent.jsm
+++ b/toolkit/components/extensions/ExtensionContent.jsm
@@ -314,23 +314,25 @@ class ExtensionContext extends BaseConte
       // This is an iframe with content script API enabled and its principal should be the
       // contentWindow itself. (we create a sandbox with the contentWindow as principal and with X-rays disabled
       // because it enables us to create the APIs object in this sandbox object and then copying it
       // into the iframe's window, see Bug 1214658 for rationale)
       this.sandbox = Cu.Sandbox(contentWindow, {
         sandboxPrototype: contentWindow,
         wantXrays: false,
         isWebExtensionContentScript: true,
+        addonId: this.extensionId,
       });
     } else {
       this.sandbox = Cu.Sandbox(prin, {
         sandboxPrototype: contentWindow,
         wantXrays: true,
         isWebExtensionContentScript: true,
         wantGlobalProperties: ["XMLHttpRequest"],
+        addonId: this.extensionId,
       });
     }
 
     let delegate = {
       getSender(context, target, sender) {
         // Nothing to do here.
       },
     };
